src/ORM/Behavior/TranslateBehavior.php

```diff
@@ -272,6 +272,13 @@ public function beforeSave(Event $event, EntityInterface $entity, ArrayObject $o
         $newOptions = [$this->_translationTable->getAlias() => ['validate' => false]];
         $options['associated'] = $newOptions + $options['associated'];
 
+        // Check early if empty translations are present in the entity.
+        // If this is the case, unset them to prevent persistence.
+        // This only applies if $this->_config['allowEmptyTranslations'] is false
+        if ($this->_config['allowEmptyTranslations'] === false) {
```

Comment:
I'm assuming, that `allowEmptyTranslations ` isn't used heavily. Doing this check in the `_unsetEmptyFields ()` would then always do an unneeded method call.